cpp/log/cms_verifier.cc

```diff
@@ -44,6 +45,75 @@ util::StatusOr<bool> CmsVerifier::IsCmsSignedByCert(BIO* cms_bio_in,
 }
 
 
+util::StatusOr<bool> CmsVerifier::IsCmsSignedByCert(const string& cms_object,
+                                                     const Cert* cert) const {
+  CHECK_NOTNULL(cert);
+
+  if (!cert->IsLoaded()) {
+    LOG(ERROR) << "Can't check cert signer as it's not loaded";
+    return Status(util::error::FAILED_PRECONDITION, "Cert not loaded");
+  }
+
+  // Load a source bio with the CMS signed data object and parse it
+  BIO* source_bio = BIO_new(BIO_s_mem());
+  BIO_write(source_bio, cms_object.c_str(), cms_object.length());
+
+  CMS_ContentInfo* const cms_content_info = d2i_CMS_bio(source_bio, nullptr);
```

Comment:
Man, we really need some `unique_ptr` typedefs/wrappers for the openssl types...